code/modules/vehicles/mecha/equipment/tools/radio.dm

```diff
@@ -0,0 +1,47 @@
+///Mech radio module
+/obj/item/mecha_parts/mecha_equipment/radio
+	name = "mounted radio"
+	desc = "A basic component of every vehicle."
+	icon_state = "mecha_radio"
+	equipment_slot = MECHA_UTILITY
+	/// Typepath of radio item
+	var/obj/item/radio/mech/radio
+
+/obj/item/mecha_parts/mecha_equipment/radio/Initialize(mapload)
+	. = ..()
+	radio = new(src)
```

Comment:
Maybe register the `COMSIG_QDELETED` signal on the radio too so our reference is cleared, just so that it does not hard delete in case some weird circumstance causes it to be deleted.